Automattic/wp-calypso
# Store Dashboard: Refactor, Update widgets to use DashboardWidget

This PR has a lot of turnover, but it boils down to three basic changes:

- Update all dashboard widgets to use `DashboardWidget` & `DashboardWidgetRow`
- Move the `setup-*` files into a `setup/` folder, to make `app/dashboard` a little easier to scan
- Remove the arrow functions where they aren't needed (on react lifecycle functions)

I also moved the `DashboardWidgetRow` component & CSS into the `dashboard-widget` folder, and deleted the now-unused `BasicWidget` & `WidgetGroup` components.

There are some minor visual changes with the switch to the standardized `DashboardWidget`:

| Empty store | Has orders |
| --- | --- |
| ![empty-fullpage](https://user-images.githubusercontent.com/541093/36746806-3798af14-1bc1-11e8-8778-b2d2ad7e5921.png) | ![has-orders-no-reviews-fullpage](https://user-images.githubusercontent.com/541093/36746808-37c27358-1bc1-11e8-8a82-11be8ae57f6f.png) |

| Has orders & has pending reviews | Not a supported location |
| --- | --- |
| ![has-orders-has-reviews-fullpage](https://user-images.githubusercontent.com/541093/36746807-37b2726e-1bc1-11e8-9b20-9be06f2ecda2.png) | ![unsupported-fullpage](https://user-images.githubusercontent.com/541093/36746809-37d65814-1bc1-11e8-9ee7-f29c63edfc98.png) |

| Medium-sized screen | Small screen |
| --- | --- |
| <img width="528" alt="screen shot 2018-02-27 at 2 03 38 pm" src="https://user-images.githubusercontent.com/541093/36748861-2159f75c-1bc7-11e8-9445-eea08ccae3ac.png"> | <img width="386" alt="screen shot 2018-02-27 at 2 03 58 pm" src="https://user-images.githubusercontent.com/541093/36748860-2147543a-1bc7-11e8-8b7e-ba0e6dbdcd31.png"> |


**To test**

- Make sure the setup flow still works, either by starting a new site or resetting your preferences in the wp.com console
- There should also be no change to the checklist
- View the dashboard: `/store/:site`
- There are different views based on your store, the states to test are:
  - No content (orders or reviews)
  - Has orders, but no pending reviews
  - Has orders, and pending reviews
  - Store is in an unsupported country
